### DISCUSSION TITLE
fatal: http://IP_of_server:8080/amitkumar10591/git-proxy.git/info/refs not valid: is this a git repository?

### DISCUSSION THREAD

**Discussion by amitkumar10591** [finos/git-proxy#490] [Q&A]
Hi Team,
I don't know if this is the right place for this question. But while configuring it I'm getting the below error:
fatal: http://IP_of_server_hosting_git-proxy:8080/amitkumar10591/git-proxy.git/info/refs not valid: is this a git repository?

Do you have any idea on how to resolve it?

**Comment by JamieSlome** [comment]
Hello @amitkumar10591 - thank you for your question 👍 Can you confirm that you see both the proxy and server running on ports `8000` and `8080`?

**Reply by amitkumar10591** [reply]
Hi Jamie,
So when I hit the URL on the browser I get the below response:
http://IP_of_server_hosting_git-proxy:8000
Error: getaddrinfo ENOTFOUND github.com
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns .js:67:26)

http://IP_of_server_hosting_git-proxy:8080
{"healthcheck":"/api/v1/healthcheck","push":"/api/v1/push","auth":"/api/v1/auth"}

**Reply by amitkumar10591** [reply]
Also when I run the npx -- @finos/git-proxy --config ../config/proxy.config.json I get the below response:
Listening on 8000
Service Listening on 8080

**Reply by amitkumar10591** [reply]
Hi @JamieSlome ,
Were you able to get a chance to look at the issue. If you need any more details then let me know.

**Reply by amitkumar10591** [reply]
Hi @JamieSlome / Team,
Can anyone update here please

**Reply by JamieSlome** [reply]
@amitkumar10591 - apologies for the delay. I've been on holiday recently. I will get back to you tomorrow 🤝

**Reply by mcleo-d** [reply]
Hi @amitkumar10591 and @JamieSlome 

Both me and @msagi are now part of the discussion and can help replicate the issue from the inside with @amitkumar10591. 

Thank you for your engagement as we get the end to end running 😁 

James.

**Reply by mcleo-d** [reply]
@amitkumar10591 and @JamieSlome 

I have looked up the following error on various `node.js` related threads and there are two potential issues that are leading to the error below.

```
Error: getaddrinfo ENOTFOUND github.com
at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns .js:67:26)
```

1. @JamieSlome can you recommend the version of node needed to run Git Proxy so we can make sure we have an adequate version running on our test server as we could be running behind. 
2. @amitkumar10591 it could be Git Proxy is unable to connect with a DNS server and can't resolve the `github.com` url. It's worth investigating whether there is a Git Proxy / local DNS config setting that could be interfering at this part of the application.

I hope this help.

James.

**Reply by msagi** [reply]
Hi folks, this is due to GitProxy is unable to resolve DNS, nothing to do with GitProxy, it is about the underlying OS does not have DNS resolution set up. I can give you more details via a call.

**Reply by maoo** [reply]
FYI - added topic to the agenda for monday's meeting

**Comment by JamieSlome** [comment]
Has this issue been resolved? @amitkumar10591

**Reply by msagi** [reply]
hi @JamieSlome , I'm happy to take this :)

**Reply by JamieSlome** [reply]
@msagi - that would be amazing, thank you ❤️

**Comment by msagi** [comment]
@mcleo-d @JamieSlome @amitkumar10591 

Lines
```
# src/proxy/processors/push-action/pullRemote.js:22..27
    const cmd = `git clone ${action.url} --bare`;
    step.log(`Exectuting ${cmd}`);

    const response = execSync(cmd, {
      cwd: action.proxyGitPath,
    }).toString('utf-8');
```

the git command needs DNS resolution to be able to resolve the git repository URL. The error message suggests, the machine runs GitProxy does not have DNS resolution enabled.

You can verify this via
```
nslookup github.com
```

**Comment by JamieSlome** [comment]
@amitkumar10591 - has this been resolved?
